src/Functions/generateUUIDv7.cpp

```diff
@@ -0,0 +1,373 @@
+#include <DataTypes/DataTypeUUID.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionsRandom.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+constexpr auto bits_in_counter = 42;
+constexpr uint64_t counter_limit = (1ull << bits_in_counter);
+constexpr uint8_t random_data_offset = 6;
+constexpr uint8_t random_data_count = 10;
+constexpr uint8_t next_count_random_data_offset = 12;
+constexpr uint8_t next_count_random_data_count = 4;
+
+using UUIDAsArray = std::array<uint8_t, 16>;
+
+uint64_t getTimestampMs()
+{
+    timespec tp;
+    clock_gettime(CLOCK_REALTIME, &tp);
+    const uint64_t sec = tp.tv_sec;
+    return sec * 1000 + tp.tv_nsec / 1000000;
+}
+
+void fillTimestamp(UUIDAsArray & uuid, uint64_t timestamp)
+{
+    uuid[0] = (timestamp >> 40) & 0xFF;
+    uuid[1] = (timestamp >> 32) & 0xFF;
+    uuid[2] = (timestamp >> 24) & 0xFF;
+    uuid[3] = (timestamp >> 16) & 0xFF;
+    uuid[4] = (timestamp >> 8) & 0xFF;
+    uuid[5] = (timestamp)&0xFF;
+}
+}
+
+#define DECLARE_SEVERAL_IMPLEMENTATIONS(...) \
```

Comment:
SIMD-fication probably makes sense for `generatUUIDv4` but it is overkill for `generateUUIDv7` as locking and retrieving the system time will be the bottlenecks. Would it be possible to get rid of all macro magic for SIMD (search for "TargetSpecific", "USE_MULTITARGET_CODE", "DECLARE_SEVERAL_IMPLEMENTATIONS" etc., also the comment on l. 217 will be obsolete)?